protocols/grpc/agent.proto

```diff
@@ -434,3 +435,25 @@ message StringUser {
 	string gid = 2;
 	repeated string additionalGids = 3;
 }
+
+message CopyFileRequest {
+	// Path is the destination file in the guest. It must be absolute,
+	// canonical and below /run.
+	string path = 1;
+	// FileSize is the expected file size, for security reasons write operations
+	// are made in a temporary file, once it has the expected size, it's moved
+	// to the destination path.
+	int64 file_size = 2;
+	// FileMode is the file mode.
+	uint32 file_mode = 3;
+	// DirMode is the mode for the parent directories of destination path.
+	uint32 dir_mode = 4;
+	// Uid is the numeric user id.
+	int32 uid = 5;
```

Comment:
There is the question of how the client is going to know the `uid` and `gid` values of course. They can prolly safely assume the standard users exist, but if they want to get elaborate, we may need to consider expanding `GetGuestDetails()` to return a list of uids+gids (optionally).... I know! This appears to be such a simple feature, but... :smile:

Reply:
yes, I think so :smiley: